atc/api/present/build.go

```diff
@@ -32,6 +33,19 @@ func Build(build db.Build) atc.Build {
 		CreatedBy:            build.CreatedBy(),
 	}
 
+	showComments := false
+	if job != nil {
+		showComments = showComments || job.Public()
+	}
+	if access != nil {
+		showComments = showComments || access.IsAuthorized(build.TeamName())
+	}
+
+	if showComments {
+		comment := build.Comment()
+		atcBuild.Comment = comment
+	}
```

Comment:
Could we add some tests for this, perhaps from the perspective of one of the endpoints that calls it. e.g. calling `GetJobBuild`, where you have access to view the build but not the comment